client/console.py

```diff
@@ -201,45 +215,56 @@ async def handle_message(self) -> str:
 
     async def register(self, username: str) -> str:
         """Sends an init request to the server to initialize our player."""
-        request = {"type": "init", "username": username}
-        await self.main_app.websocket.send(json.dumps(request))
+        request = InitializePlayer(type="init", username=username)
+        await self.main_app.websocket.send(request.json())
         self.initialized = True
 
-        return ""
+        return NOTHING_TO_LOG
 
     @enforce_initialization
     async def send_chat_message(self) -> str:
-        response = json.dumps(
-            {
-                "type": "chat",
-                "player_name": self.main_app.name,
-                "chat_message": self.message,
-            }
+        response = ChatMessage(
+            type="chat",
+            # this function will only run when the client has registers, so self.main_app.name will never be None
```

Comment:
Let's put an assert instead -- this way we have our invariants clearly visible and get an error if they're violated.